File: BE/src/main/java/com/example/beprojectweb/entity/Category.java
package com.example.beprojectweb.entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    int cate_ID;
    String name;
    String description;
    @OneToMany(mappedBy = "category", cascade = CascadeType.ALL)
    @JsonManagedReference
    List<Product> productList;
}

File: FE/src/model/Category.ts
import type { Product } from "./Product";

export interface Category {
  cate_ID: number;
  name: string;
  description: string;
  productList: Product[]; 
}

export interface CategoryRequest {
    name: string;
    description: string;
  }

export interface CategoryResponse {
  cate_ID: number;
  name: string;
  description: string;
  productList: Product[];
}